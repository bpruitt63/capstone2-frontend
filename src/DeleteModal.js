import React, { useState } from 'react';
import './static/styles/Trips.css';
import { Button, Modal, ModalBody, ModalFooter } from 'reactstrap';


/** Modal to confirm deletion of a saved trip */
function DeleteModal({deleteTrip, username, id}) {

    const [modal, setModal] = useState(false);

    const toggle = () => setModal(!modal);

    return (
        <div>
            <Button color="danger" className='toggle' onClick={toggle}>
                Delete Trip
            </Button>
            <Modal isOpen={modal} toggle={toggle}>
                <ModalBody>
                    Are you sure you want to permanently delete this trip?
                </ModalBody>
                <ModalFooter>
                    <Button color="danger" 
                            onClick={() => deleteTrip(username, id)}>
                        Delete
                    </Button>{' '}
                    <Button onClick={toggle}>Cancel</Button>
                </ModalFooter>
            </Modal>
        </div>
    );
};

export default DeleteModal;