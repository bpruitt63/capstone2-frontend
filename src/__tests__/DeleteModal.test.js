import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import DeleteModal from '../DeleteModal';


test('renders', () => {
    render(<DeleteModal/>);
    expect(screen.queryByText(/Are you sure you want to permanently delete this trip/)).not.toBeInTheDocument;
    const toggleButton = screen.getByText(/Delete Trip/);
    fireEvent.click(toggleButton);
    expect(screen.getByText(/Are you sure you want to permanently delete this trip/)).toBeInTheDocument;
});

test('matches snapshot', () => {
    const {asFragment} = render(<DeleteModal/>);
    expect(asFragment()).toMatchSnapshot();
});