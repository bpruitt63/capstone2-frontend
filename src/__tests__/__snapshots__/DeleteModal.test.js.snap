// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <button
      class="toggle btn btn-danger"
      type="button"
    >
      Delete Trip
    </button>
  </div>
</DocumentFragment>
`;
